src/dev-server/index.ts

```diff
@@ -54,18 +57,27 @@ export default class DevServer {
       process.stdin.resume()
       process.stdin.setEncoding('utf8')
       process.stdin.on('data', async (key: string) => {
-        if (key === CTRL_C) resolve()
-        if (key === 'b') await this.rebuild()
-        if (key === 't') await this.terminateAllDevbackends()
+        try {
+          if (key === CTRL_C) resolve()
+          if (key === 'b') await this.rebuild()
+          if (key === 't') await this.terminateAllDevbackends()
+        } catch {
+          resolve()
+        }
       })
-    })
 
-    let fsWatcher: chokidar.FSWatcher | null = null
-    if (watch) {
-      this.updateFooterAndLog([`Watching ${watch.join(', ')} for changes...`, ''])
-      const watchPaths = watch.map(w => path.resolve(process.cwd(), w))
-      fsWatcher = chokidar.watch(watchPaths).on('change', this.rebuild.bind(this))
-    }
+      if (watch) {
+        this.updateFooterAndLog([`Watching ${watch.join(', ')} for changes...`, ''])
+        const watchPaths = watch.map(w => path.resolve(process.cwd(), w))
+        this.fsWatcher = chokidar.watch(watchPaths).on('change', async () => {
```

Comment:
does anything prevent this from being run twice in parallel (if files are changed in quick succession)?

Reply:
Good question - it's not strictly enforced, but the `buildImage()` function is blocking so it happens that images won't be built at the same time. We could debounce, though if the `buildImage()` function became async, debouncing wouldn't prevent parallel builds. The "right" approach might be to make `buildImage()` async and cancellable, but I think that grows the scope of this PR too much. The dev CLI will change soon anyway (given customer feedback).